evm/src/witness/transition.rs

```diff
@@ -250,6 +255,11 @@ fn try_perform_instruction<F: Field>(state: &mut GenerationState<F>) -> Result<(
     row.stack_len = F::from_canonical_usize(state.registers.stack_len);
 
     let opcode = read_code_memory(state, &mut row);
+    (row, opcode)
+}
+
+fn try_perform_instruction<F: Field>(state: &mut GenerationState<F>) -> Result<(), ProgramError> {
+    let (mut row, opcode) = base_row(state);
     let op = decode(state.registers, opcode)?;
 
```

Comment:
Somewhat related to this PR, is the `assert_eq!(rhs, F::ZERO)` below on line 286 necessary? It makes a bunch of tests fail (e.g. `stStackTests/stackOverflowDUP_0`) and removing it doesn't seem to cause any other issues.

Reply:
Hmmmm, if that assertion doesn't hold, then the constraints _should_ fail. But since they don't, then there's some kind of bug. I've been meaning to simplify this part a bit anyway, so I'll just rewrite it.